### PR TITLE
Extract events from MainSummary into a separate dataset

### PR DESCRIPTION
**Pull request by sunahsuh**
Adds a job to extract events from main summary into an Events dataset (under a `doctype=main` partition), tests for the job and initial documentation.

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/views/MainEventsView.scala [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    exploded.selectExpr("document_id", "client_id", "normalized_channel", "country", "locale", "app_name",
+      "app_version", "os", "os_version", "e10s_enabled", "e10s_cohort", "subsession_start_date", "subsession_length",
+      "sync_configured", "sync_count_desktop", "sync_count_mobile", "timestamp", "sample_id",
+      // Flatten nested event fields.
+      "events.timestamp as event_timestamp",
+      "events.category as event_category",
+      "events.method as event_method",
+      "events.object as event_object",
+      "events.string_value as event_string_value",
+      "events.map_values as event_map_values")
```
Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainEventsView.scala]: Is the client guaranteed to send `map_values` as `Map[String,String]`? Does it need error handling or coercion to strings?

File: src/test/scala/com/mozilla/telemetry/MainEventsViewTest.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            e.copy(timestamp = 123))))
+      ).toDS().toDF()
+
+      pings.count() should be(4)
+      val events = MainEventsView.eventsFromMain(pings, None)
+
+      events.count() should be(4)
+
+      events.select("client_id").distinct().count() should be(2)
+      events.select("document_id").distinct().count() should be(3)
```
Comment (mreid-moz) [inline src/test/scala/com/mozilla/telemetry/MainEventsViewTest.scala]: Please add a test that covers the 'sample id' filtering option of `eventsFromMain`

File: src/main/scala/com/mozilla/telemetry/views/MainEventsView.scala (hunk 1 above)
Comment (sunahsuh) [inline src/main/scala/com/mozilla/telemetry/views/MainEventsView.scala]: Should be addressed by #175

File: src/main/scala/com/mozilla/telemetry/views/MainEventsView.scala [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    val eventsSchema = MainSummaryView.buildEventSchema
+    val partialDataFrame = mainSummaryData
+      .select("document_id", "client_id", "normalized_channel", "country", "locale", "app_name", "app_version", "os",
+        "os_version", "e10s_enabled", "e10s_cohort", "subsession_start_date", "subsession_length", "sync_configured",
+        "sync_count_desktop", "sync_count_mobile", "timestamp", "sample_id", "events")
+      .where("client_id is not null")
+      .where("events is not null")
+
+    val records = sampleId match {
+      case Some(s) => partialDataFrame.where(s"sample_id like $s")
```
Comment (sunahsuh) [inline src/main/scala/com/mozilla/telemetry/views/MainEventsView.scala]: 'Twas habit, but `=` is probably faster

**Comment by sunahsuh** [comment]
RFAL

**Comment by mreid-moz** [comment]
Please rebase, then we should be good to go.

**Pull request merged**